include/experimental/__p2642_bits/layout_padded.hpp

```diff
@@ -0,0 +1,742 @@
+{
+  if ( __alignment == 0 )
+    return _T(0);
+  else
+    return ( ( __offset + __alignment - 1 ) / __alignment) * __alignment;
+}
+
+template<class _ExtentsType, size_t _PaddingStride, size_t _ExtentToPadRank>
+MDSPAN_INLINE_FUNCTION
+constexpr size_t
+__get_actual_padding_stride()
+{
+  constexpr auto __rank = _ExtentsType::rank();
+
+  if constexpr (__rank <= size_t(1)) {
+    return _PaddingStride;
+  } else if constexpr (_PaddingStride != dynamic_extent &&
+                       _ExtentsType::static_extent(_ExtentToPadRank) != dynamic_extent) {
+    static_assert((_PaddingStride != 0) || (_ExtentsType::static_extent(_ExtentToPadRank) == 0), "padding stride can be 0 only if extents_type::static_extent(extent-to-pad) is 0");
+    return __find_aligned_offset(_PaddingStride, _ExtentsType::static_extent(_ExtentToPadRank));
+  } else {
+    return dynamic_extent;
+  }
+}
+
+template<size_t _ExtentToSub, class _Extents, size_t _NewExtent, class _Indices>
+struct __substitute_extents_impl;
+
+template<size_t _ExtentToSub, class _IndexType, size_t... _OrigExtents, size_t _NewExtent, size_t... _Indices>
+struct __substitute_extents_impl<_ExtentToSub, extents<_IndexType, _OrigExtents...>, _NewExtent, index_sequence<_Indices...>>
+{
+  using __orig_extents_type = extents<_IndexType, _OrigExtents...>;
+  using __type = extents<_IndexType, ((_Indices == _ExtentToSub) ? _NewExtent : _OrigExtents)...>;
+
+  template <typename _T>
+  static constexpr auto
+  __construct_with_type(const __orig_extents_type &__extents, const extents<_IndexType, _NewExtent> &__new_extents)
+  {
+    return _T{((_Indices == _ExtentToSub) ? __new_extents.extent(0) : __extents.extent(_Indices))...};
+  }
+
+  MDSPAN_INLINE_FUNCTION
+  static constexpr auto
+  __construct(const __orig_extents_type &__extents, const extents<_IndexType, _NewExtent> &__new_extents)
+  {
+    return __construct_with_type<__type>(__extents, __new_extents);
+  }
+
+  MDSPAN_INLINE_FUNCTION
+  static constexpr auto
+  __construct(const __orig_extents_type &__extents)
+  {
+    return __type{__extents.extent(_Indices)...};
+  }
+};
+
+template<size_t _ExtentToSub, class _Extents, size_t _NewExtent>
+using __substitute_extents = __substitute_extents_impl<_ExtentToSub, _Extents, _NewExtent, make_index_sequence<_Extents::rank()>>;
+
+template<size_t _ExtentToPadRank, class _Extents, size_t _ActualPaddingStride, class _Enabled = void>
+struct __inner_extents_type
+{
+  using __subs_type = __substitute_extents< _ExtentToPadRank, _Extents, _ActualPaddingStride >;
+  using __type = typename __subs_type::__type;
+
+  template<size_t _PaddingStride>
+  MDSPAN_INLINE_FUNCTION
+  static constexpr auto
+  __construct(const _Extents &__extents)
+  {
+    if constexpr (_PaddingStride == dynamic_extent)
+    {
+      return __subs_type::__construct(__extents);
+    } else {
+      // This is a corner case where `_PaddingStride` is not dynamic but `_ActualPaddingStride` is
```

Comment:
I would expect this to be a common case, rather than a "corner case."  Many users will use the padding stride to express an overalignment factor for SIMD or some other hardware reason, but will combine that with a dynamic extent.

Reply:
I changed the comment to just read "case"